docs/source/metrics.rst

```diff
@@ -4,7 +4,16 @@ ignite.metrics
 Metrics provide a way to compute various quantities of interest in an online
 fashion without having to store the entire output history of a model.
 
-In practice a user needs to attach the metric instance to an engine. The metric
+Attach Engine API
+------------------
+
+The metrics as stated above are computed in a online fashion,which means that the metric instance accumulates some internal counter on 
+each iteration and accuracy value is computed once epoch is ended. Internal counters are reset after every epoch.
+For doing this we use three public APIs ``reset`` , ``update`` and ``compute`` :ref:`read about them here<Reset, Update, Compute API>`. We internally accumulate some counters
+when there is a update call. The metric's value is computed on each compute call and counters are reset on each reset
+call. 
```

Comment:
This part seems to repeat the above text without bringing more info.

Reply:
@vfdev-5 Should I delete the entire portion or trim it ?